internal/clients/nightfall/nightfall.go

```diff
@@ -250,45 +286,71 @@ func (n *Client) scanContent(
 	}
 
 	// Determine findings from response and create comments
-	createdComments := createCommentsFromScanResp(cts, resp, n.TokenExclusionList)
+	createdComments := createCommentsFromScanRespForFiles(cts, resp, n.TokenExclusionList)
 	logger.Info(fmt.Sprintf("Got %d annotations for request #%d", len(createdComments), requestNum))
 	return createdComments, nil
 }
 
-func (n *Client) scanAllContent(
+func (n *Client) scanAllFiles(
 	ctx context.Context,
 	logger logger.Logger,
-	cts []*contentToScan,
+	cts []*fileToScan,
 	commentCh chan<- []*diffreviewer.Comment,
 ) {
 	defer close(commentCh)
 	blockingCh := make(chan struct{}, n.MaxNumberRoutines)
 	var wg sync.WaitGroup
+	requestBatches := make([][]*fileToScan, 0, 1)
+	endIndex := 0
+	for endIndex < len(cts) {
+		requestBatch := make([]*fileToScan, 0)
+		startIndex := endIndex
+		currentSize := 0
+		for {
+			if endIndex >= len(cts) {
```

Comment:
Feel like we might want to separate this into 2 different cases. 
`if endIndex == len(cts) {break}`
`if endIndex > len(cts) {//something bad happend}`